node-api/backend/validator.go

```diff
@@ -21,11 +21,75 @@
 package backend
 
 import (
+	"slices"
+
 	"github.com/berachain/beacon-kit/node-api/backend/utils"
 	beacontypes "github.com/berachain/beacon-kit/node-api/handlers/beacon/types"
 	"github.com/berachain/beacon-kit/primitives/math"
 )
 
+// FilteredValidators will grab all of the validators from the state at the
+// given slot. It will then filter them by the provided ids and statuses.
+func (b Backend) FilteredValidators(
+	slot math.Slot, ids []string, statuses []string,
+) ([]*beacontypes.ValidatorData, error) {
+	st, _, err := b.stateFromSlot(slot)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert requested ids (can be validator index or pubkey) to validator index only.
+	validatorIndicies := make([]uint64, 0, len(ids))
+	for _, id := range ids {
+		validatorIndex, vErr := utils.ValidatorIndexByID(st, id)
```

Comment:
I wonder if we should treat pubKey and index cases differently? This seems wasterful
If we have pubKeys, we don't need to do:
- pubKey -> index
- validator -> pubKey -> index -> check if the index is in the required list

We can load the validators directly by pubKey and then retrieve their index to be serialized in the response

Reply:
I have removed the logic where we are getting indexes before getting all the validators. Tried to handle it differently in `FilteredValidators` by first iterating over public key.